runtime/src/iree/hal/buffer.c

```diff
@@ -280,7 +280,10 @@ IREE_API_EXPORT iree_status_t iree_hal_buffer_validate_memory_type(
 IREE_API_EXPORT iree_status_t iree_hal_buffer_validate_access(
     iree_hal_memory_access_t allowed_memory_access,
     iree_hal_memory_access_t required_memory_access) {
-  if (iree_all_bits_set(allowed_memory_access, IREE_HAL_MEMORY_ACCESS_ANY)) {
+  // TODO(scotttodd): change to allowed_memory_access when possible
+  //   (at least the web samples aren't setting the required access bits)
+  // if (iree_all_bits_set(allowed_memory_access, IREE_HAL_MEMORY_ACCESS_ANY)) {
+  if (iree_all_bits_set(required_memory_access, IREE_HAL_MEMORY_ACCESS_ANY)) {
```

Comment:
I saw the `"memory access must specify one or more of _READ or _WRITE` error just below this running https://github.com/iree-org/iree/tree/main/experimental/web/sample_dynamic (CPU code), but I also see this error running with WebGPU:

`iree_api_webgpu.js:76 (C) iree/runtime/src/iree/hal/command_buffer_validation.c:80: PERMISSION_DENIED; requested buffer usage is not supported for the buffer on this queue; buffer allows TRANSFER|MAPPING, operation requires TRANSFER (allocator compatibility mismatch)`

Both will need to be fixed somehow?